edgelet/iotedged/src/error.rs

```diff
@@ -25,6 +31,15 @@ pub enum ErrorKind {
     #[fail(display = "The daemon could not start up successfully: {}", _0)]
     Initialize(InitializeErrorReason),
 
+    #[fail(display = "Invalid device configuration was provided.")]
+    InvalidDeviceConfig,
+
+    #[fail(display = "Invalid IoT hub configuration was provided.")]
+    InvalidHubConfig,
+
+    #[fail(display = "Invalid signed token was provided.")]
+    InvalidSignedToken,
+
```

Comment:
We should consider moving these into `InitializeErrorReason`s or create a `Watchdog(WatchdogErrorReason)` variant, instead of adding these to the flat namespace.

Reply:
@arsing do you have any comments on the error handling?

Reply:
I agree for the hub and device config but the signed tokens can be invalid after successful initialization too i.e., the daemon can get 401 errors from IoT hub after initialization too. Should we keep that error type in the flat namespace then?